### PR TITLE
Feat: rework gcode handling

### PR DESCRIPTION
**Pull request by andrushechka-b**
The main idea is that we do a check for a cone path by U coordinate value.
If U coordinate differs from the current bed plane Z rotation, then the path is cone.

This pr also speeds up gcode loading, without this pr loading time is 20% higher

### CODE REVIEW

File: src/gcode.py [hunk 1]
```diff
@@ -253,50 +276,38 @@ def finishLayer():
...
-                    path.append(Point(x, y, z, a, b))
-                else:
-                    # U changed, get interpolated points
-                    for point in printer.getPoints():
-                        x, y, z, a, b = parseArgs(point, x, y, z, a, b, cone_axis, rotationPoint, abs_pos)
-                        path.append(Point(x, y, z, a, b))
-
             elif args[0] == "G90":  # absolute positioning
-                abs_pos = True
+                printer.abs_pos = True
```
Comment (lvjonok) [inline src/gcode.py:299]: as I see from gcode in goosli, we do not generate in relative positioning, so there we can simplify even this

File: src/gcode.py [hunk 2]
```diff
@@ -59,116 +59,163 @@ def xyz(self, rot):
...
+            self.setAbsPos(args)
+        else:
+            self.setRelPos(args)
+
+        self.dX = self.prevPos.X - self.currPos.X
+        self.dY = self.prevPos.Y - self.currPos.Y
+        self.dZ = self.prevPos.Z - self.currPos.Z
         self.dU = self.prevPos.U - self.currPos.U
+        self.dE = self.prevPos.E - self.currPos.E
+        noMove = self.dX == 0 and self.dY == 0 and self.dZ == 0 and self.dU == 0
```
Comment (lvjonok) [inline src/gcode.py:143]: almostZero?)

**Review by lvjonok — APPROVED** [review]
I like it

File: src/gcode.py (hunk 1 above)
Comment (andrushechka-b) [reply src/gcode.py:299]: I would leave it, the implementation is not so difficult

File: src/gcode.py (hunk 2 above)
Comment (andrushechka-b) [reply src/gcode.py:143]: no,I think not needed here, we should behave like a real printer, it considers even a very slow movement as movement

Comment (andrushechka-b) [reply src/gcode.py:143]: I think goosli should care that two sequent lines in gcode differ enaugh

**Pull request merged**